heartbeat/monitors/active/http/check_test.go

```diff
@@ -80,24 +86,70 @@ func TestCheckBody(t *testing.T) {
 			second line
 			third line`,
 			[]string{"(?s)first.*fourth.*third"},
+			nil,
 			false,
 		},
 		{
 			"Single regex that doesn't match",
 			"ok",
 			[]string{"notok"},
+			nil,
 			false,
 		},
 		{
 			"Multiple regex match where at least one must match",
 			"ok",
 			[]string{"ok", "yay"},
+			nil,
 			true,
 		},
 		{
 			"Multiple regex match where none of the patterns match",
 			"ok",
 			[]string{"notok", "yay"},
+			nil,
+			false,
+		},
+		{
+			"Only positive check where pattern matches HTTP return body",
+			"'status': 'red'",
+			[]string{"red"},
+			nil,
+			true,
+		},
+		{
+			"Only positive check where pattern not match HTTP return body",
+			"'status': 'green'",
+			[]string{"red"},
+			nil,
+			false,
+		},
+		{
+			"Only negative check where pattern matches HTTP return body",
+			"'status': 'red'",
+			nil,
+			[]string{"red"},
+			false,
+		},
+		{
+			"Only negative check where pattern not match HTTP return body",
+			"'status': 'green'",
+			nil,
+			[]string{"red"},
+			true,
+		},
+		{
+			"Positive with negative check where all positive pattern matches and none negative check matches",
+			"'status': 'green', 'cluster': 'healthy'",
+			[]string{"green"},
+			[]string{"unhealthy"},
+			true,
+		},
+		{
+			"Positive with negative check where positive and negative pattern both match",
```

Comment:
Thanks for adding these tests!

Reply:
Sure, add another test in case positive and negative are both empty.

Reply:
Added the doc in `heartbeat/docs/monitors/monitor-http.asciidoc` file.
Please help to review. thx.